modules/barcode/src/decoder/common/hybrid_binarizer.cpp

```diff
@@ -0,0 +1,195 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+// Modified from ZXing. Copyright ZXing authors.
+// Licensed under the Apache License, Version 2.0 (the "License").
+
+#include "../../precomp.hpp"
+#include "hybrid_binarizer.hpp"
+
+namespace cv {
+namespace barcode {
+
+
+#define CAP(x, x1, x2) x < (x1) ? (x1) : ((x) > (x2) ? (x2) : (x))
```

Comment:
> CAP

`CLAMP`?

BTW, [`std::clamp`](https://en.cppreference.com/w/cpp/algorithm/clamp) can't be used because it is C++17 only